### PR TITLE
feat(swap): plumb affiliate to THORChain + Mayachain quote URLs

### PR DESCRIPTION
**Pull request by gomesalexandre**
Stacked on #590 - that PR should be merged first.

## what

Conditionally appends `affiliate` + `affiliate_bps` query params to the
THORChain and Mayachain `/quote/swap` URL builders when the caller
provides a non-nil, positive `AffiliateBps` and a non-empty
`AffiliateAddress`. Both providers use identical param names (Maya is a
THORChain fork).

## why

THORChain and Mayachain are the highest-revenue swap providers. The base
affiliate fields (`AffiliateBps`, `AffiliateAddress`) shipped in #590 -
this is the downstream wire-up that makes those fields actually land in
the upstream quote URLs. Without this, affiliate fees are silently
ignored even after the calling layer starts passing them.

## how

- `sdk/swap/thorchain.go`: after the `tolerance_bps` param, add a guard
  `if req.AffiliateBps != nil && *req.AffiliateBps > 0` - if the address is
  also non-empty, set both params; if address is empty, `log.Printf` and
  skip (no auto-routing to an unknown treasury).
- `sdk/swap/mayachain.go`: identical pattern (same URL params, same
  guard semantics).
- `sdk/swap/affiliate_test.go`: 8 unit tests (4 per provider) covering
  nil bps, zero bps, positive bps + empty address, and positive bps +
  non-empty address. Uses `httptest.NewServer` stubs that capture emitted
  URL params without hitting a live node.

## risk

Low - conditional on non-nil `AffiliateBps`. Existing callsites pass
`nil` by default so the URL output is unchanged. Only 2 files touched.

## receipts

```
$ go build ./sdk/...
(clean)

$ go vet ./sdk/...
(clean)
```

Note: `go test ./sdk/swap/...` fails to link due to a pre-existing
`bytedance/sonic` asm compatibility issue on Go 1.26 (the `replace`
directive in `go.mod` pins sonic v1.12.0 which references internal
`encoding/json` symbols removed in 1.26). This is present on the base
branch before any of our changes - confirmed on both `main` and
`feat_recipes_affiliate_quoterequest`. The 8 unit tests are written and
will pass once the sonic pin is updated or the build environment matches.

Closes vultisig-ops bead W2-2 (vultisig-ops-4 in beads-local).

🤖 Agent-generated

### CODE REVIEW

File: sdk/swap/thorchain.go [hunk 1]
```diff
@@ -210,6 +212,18 @@ func (p *THORChainProvider) GetQuote(ctx context.Context, req QuoteRequest) (*Qu
 	}
 	params.Set("tolerance_bps", fmt.Sprintf("%d", toleranceBps))
 
+	// Plumb affiliate fee when caller has both a positive bps and a non-empty address.
+	// nil or zero bps -> skip (preserves backward compat with existing callsites).
+	// positive bps + empty address -> skip silently; don't auto-route to an unknown address.
+	if req.AffiliateBps != nil && *req.AffiliateBps > 0 {
+		if req.AffiliateAddress == "" {
```
Comment (0xApotheosis) [inline sdk/swap/thorchain.go:222]: This skips the affiliate when the address is empty, but `QuoteRequest` says empty means use the provider/chain default treasury. Can we either resolve that default here or make a non-empty address required with positive bps?

File: sdk/swap/mayachain.go [hunk 2]
```diff
@@ -158,6 +160,18 @@ func (p *MayachainProvider) GetQuote(ctx context.Context, req QuoteRequest) (*Qu
 	params.Set("streaming_interval", "3")
 	params.Set("streaming_quantity", "0")
 
+	// Plumb affiliate fee when caller has both a positive bps and a non-empty address.
+	// nil or zero bps -> skip (preserves backward compat with existing callsites).
+	// positive bps + empty address -> skip silently; don't auto-route to an unknown address.
+	if req.AffiliateBps != nil && *req.AffiliateBps > 0 {
```
Comment (0xApotheosis) [inline sdk/swap/mayachain.go:166]: Can we validate Maya's affiliate fee range before forwarding this? The Maya docs cap affiliate fees at 500 bps, so larger values become upstream quote failures/fallbacks instead of local input errors.

**Review by 0xApotheosis — CHANGES_REQUESTED** [review]
The focused affiliate tests pass locally with `CGO_ENABLED=0 GOMODCACHE=/tmp/recipes-pr591-gomodcache GOCACHE=/tmp/recipes-pr591-gocache go test ./sdk/swap -run 'Affiliate' -count=1`. Full `go test`/`go build`/`go vet ./sdk/...` is blocked on this host by disk exhaustion after retrying with a writable module cache.

A couple contract issues below before this lands.

**Review by NeOMakinG — CHANGES_REQUESTED** [review]
## Verdict: REQUEST_CHANGES

Reviewed with GLM-5.1 engine. The affiliate plumbing logic is sound and well-tested, but there are **two contract issues** that must be addressed before merge.

---

### Blocker 1: Type Contract vs Implementation Mismatch

**Location:** `sdk/swap/types.go:128-132` vs `sdk/swap/thorchain.go:217-218`

**The Issue:**
The type comment states:
```go
// AffiliateAddress is the on-chain destination for affiliate fees. Empty
// means use the provider/chain default treasury (resolved per-chain in
// the affiliate chain maps). Non-empty overrides the default.
```

But the implementation:
```go
if req.AffiliateAddress == "" {
    log.Printf("thorchain: AffiliateBps=%d but AffiliateAddress is empty - skipping affiliate params", *req.AffiliateBps)
} else {
    params.Set("affiliate", req.AffiliateAddress)
    params.Set("affiliate_bps", strconv.Itoa(*req.AffiliateBps))
}
```

**The Problem:**
When `AffiliateAddress` is empty, the code **skips the affiliate params entirely**, but the type contract says "use the provider/chain default treasury." Callers reading the type would expect empty address to route to a default treasury, but nothing gets routed.

**Fix Options:**
1. **Implement default treasury resolution**: Add a map of default treasury addresses per-chain and use them when `AffiliateAddress` is empty
2. **Require non-empty**: Change the implementation to return an error when `AffiliateBps > 0` and `AffiliateAddress == ""`
3. **Update the contract**: Change the type comment to say "Empty means no affiliate fee" to match the current implementation

Option 3 is the smallest change for this PR scope. Options 1-2 require more design work.

---

### Blocker 2: Missing Mayachain 500 BPS Cap Validation

**Location:** `sdk/swap/mayachain.go:163-171`

**The Issue:**
Mayachain docs cap affiliate fees at 500 bps (5%). The code forwards any positive value without validation:
```go
if req.AffiliateBps != nil && *req.AffiliateBps > 0 {
    if req.AffiliateAddress == "" {
        log.Printf("mayachain: AffiliateBps=%d but AffiliateAddress is empty - skipping affiliate params", *req.AffiliateBps)
    } else {
        params.Set("affiliate", req.AffiliateAddress)
        params.Set("affiliate_bps", strconv.Itoa(*req.AffiliateBps))
    }
}
```

**The Problem:**
A caller passing `AffiliateBps: intPtr(1000)` would send `affiliate_bps=1000` to Mayachain, which will reject it at the upstream API. This converts a **local input error** (easy to debug, clear message) into an **upstream quote failure** (harder to trace, appears as "no route available").

**Fix:**
```go
if req.AffiliateBps != nil && *req.AffiliateBps > 0 {
    if req.AffiliateBps != nil && *req.AffiliateBps > 500 {
        return nil, fmt.Errorf("mayachain: affiliate_bps %d exceeds maximum 500 (5%%)", *req.AffiliateBps)
    }
    if req.AffiliateAddress == "" {
        log.Printf("mayachain: AffiliateBps=%d but AffiliateAddress is empty - skipping affiliate params", *req.AffiliateBps)
    } else {
        params.Set("affiliate", req.AffiliateAddress)
        params.Set("affiliate_bps", strconv.Itoa(*req.AffiliateBps))
    }
}
```

Note: THORChain typically caps at 1000 bps, so consider adding similar validation there for consistency.

---

### What Looks Good

✅ **Three-way guard is correct**: nil, zero, positive+empty, positive+non-empty — all cases handled
✅ **Test coverage is comprehensive**: All four combinations tested for both providers
✅ **No injection risk**: `params.Set()` handles URL encoding correctly
✅ **Code duplication is acceptable**: The affiliate block is identical in both providers; if it diverges later, the duplication will be justified

---

### Non-Blocking Follow-ups (from GLM-5.1)

1. **Add upper-bound validation for THORChain** (1000 bps cap)
2. **Consider trimming whitespace** on `AffiliateAddress` before empty check
3. **Consider structured logging** instead of `log.Printf`
4. **Add tests for edge cases**: negative bps, whitespace-only address, special characters in address

---

### Receipts

- **Reviewers**: GLM-5.1 (approve with notes), codex (unavailable - environment issue)
- **Test execution**: Skipped due to sonic library build issue in review environment; 0xApotheosis confirmed focused tests pass locally
- **Source validation**: All logic paths reviewed and correct
- **Contract blockers**: Type comment mismatch + missing Mayachain validation

---

**Decision**: Please address the two contract issues above. The affiliate plumbing itself is well-implemented.

File: sdk/swap/thorchain.go (hunk 1 above)
Comment (gomesalexandre) [reply sdk/swap/thorchain.go:222]: Noted - the docstring at sdk/swap/types.go:128-132 references a default-treasury fallback via 'affiliate chain maps' that don't exist in the recipes repo yet. Will align doc + code in a follow-up; minimum-scope path is tightening the docstring to match current behavior (empty address = no affiliate). Deferring to orchestrator decision before next push.

— vultisig-ops

File: sdk/swap/mayachain.go (hunk 2 above)
Comment (gomesalexandre) [reply sdk/swap/mayachain.go:166]: Noted but not blocking. Caller at mcp-ts execute_swap restricts to enum-validated bps in the 0-50 range; upstream Maya 4xx is the realistic worst case here. Defense-in-depth range check is a nice-to-have follow-up.

— vultisig-ops

File: sdk/swap/thorchain.go (hunk 1 above)
Comment (gomesalexandre) [reply sdk/swap/thorchain.go:222]: done in 16b978f on this branch - tightened the AffiliateAddress docstring to match actual behavior (empty = no affiliate params sent, no fee captured). No chain maps exist in the repo, so the previous 'default treasury' promise was dead code copy. #592 still inherits the stale docstring from the shared base (feat_recipes_affiliate_quoterequest) since this fix is on feat_recipes_affiliate_thor_maya; once this merges into base, #592 picks it up.

File: sdk/swap/thorchain.go [hunk 3]
```diff
@@ -210,6 +212,18 @@ func (p *THORChainProvider) GetQuote(ctx context.Context, req QuoteRequest) (*Qu
 	}
 	params.Set("tolerance_bps", fmt.Sprintf("%d", toleranceBps))
 
+	// Plumb affiliate fee when caller has both a positive bps and a non-empty address.
+	// nil or zero bps -> skip (preserves backward compat with existing callsites).
+	// positive bps + empty address -> skip silently; don't auto-route to an unknown address.
+	if req.AffiliateBps != nil && *req.AffiliateBps > 0 {
```
Comment (0xApotheosis) [inline sdk/swap/thorchain.go:218]: Can we reject values outside THORChain's documented 0-1000 bps range here instead of forwarding them to provider fallback?

File: sdk/swap/mayachain.go (hunk 2 above)
Comment (0xApotheosis) [inline sdk/swap/mayachain.go:166]: This still needs a local Maya range check; Maya caps affiliate fees at 500 bps, so larger values should fail before the quote request.

**Review by 0xApotheosis — CHANGES_REQUESTED** [review]
Still needs local bounds checks before this lands. `go test ./...`, `go vet ./...`, `go build ./sdk/...`, and `go test ./sdk/swap/...` pass locally; `gofmt -l sdk/swap/affiliate_test.go sdk/swap/mayachain.go sdk/swap/thorchain.go sdk/swap/types.go` still reports `sdk/swap/mayachain.go`.

File: sdk/swap/thorchain.go (hunk 3 above)
Comment (gomesalexandre) [reply sdk/swap/thorchain.go:218]: done in c2cd378 - rejects affiliate_bps > 1000 with a local error before the URL params are set, mirroring the existing tolerance_bps guard

File: sdk/swap/mayachain.go (hunk 2 above)
Comment (gomesalexandre) [reply sdk/swap/mayachain.go:166]: done in c2cd378 - same pattern, rejects > 500 locally before forwarding to Maya endpoints. test coverage in affiliate_test.go (BpsAbove500_Error + Bps500_Ok)
